tools/checkTheme.js

```diff
@@ -184,6 +191,79 @@ function check_group(group, rules) {
   }
 }
 
+const round2 = (x) => Math.round(x*100)/100;
+
+class CSSRule {
+  constructor(rule, body) {
+    this.rule = rule;
+    if (rule.declarations) {
+      this.bg = rule.declarations.find(x => x.property =="background")?.value;
+      this.fg = rule.declarations.find(x => x.property =="color")?.value;
+
+      if (this.bg) {
+        this.bg = csscolors[this.bg] || this.bg;
+      }
+      if (this.fg) {
+        this.fg = csscolors[this.fg] || this.fg;
+      }
+
+      // inherit from body if we're missing fg or bg
+      if (this.hasColor) {
+        if (!this.bg) this.bg = body.background;
+        if (!this.fg) this.fg = body.foreground;
+      }
+    }
+  }
+  get background() {
+    return this.bg
+  }
+
+  get foreground() {
+    return this.fg
+  }
+  get hasColor() {
+    if (!this.rule.declarations) return false;
+    return this.fg || this.bg;
+  }
+  toString() {
+    return ` ${this.foreground} on ${this.background}`
+  }
+
+  contrastRatio() {
+    if (!this.foreground) return "unknown (no fg)"
+    if (!this.background) return "unknown(no bg)"
+    return round2(wcagContrast.hex(this.foreground, this.background));
```

Comment:
We have some themes that use `rgba()`, will `.hex()` work on those values?

https://github.com/highlightjs/highlight.js/blob/0bf3adeecd5ba76cef8ee843e38722089b232da1/src/styles/arduino-light.css#L52-L54